### PR TITLE
Add new initial admin password handling

### PR DESCRIPTION
**Pull request by Jotschi**

### CODE REVIEW

**Comment by Jotschi** [comment]
* [ ] Handle test performance regression
* [ ] Add tests
* [ ] Add env flags to demo/musetech k8s deployment
* [ ] Add changelog

File: api/src/main/java/com/gentics/mesh/util/PasswordUtil.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.Random;
+
+/**
+ * Utility class which provides methods to handle passwords.
+ */
+public final class PasswordUtil {
+
+	private static final String ALPHA_NUMERIC_READABLE_CHAR = "ABCDEFGHJKLMNPQRTUVWXYZ"
+		+ "2346789"
+		+ "abcdefghjkmnpqrtuvxyz";
```
Comment (philippguertler) [inline api/src/main/java/com/gentics/mesh/util/PasswordUtil.java]: `pwgen` uses these characters:
```
const char *pw_digits = "0123456789";
const char *pw_uppers = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
const char *pw_lowers = "abcdefghijklmnopqrstuvwxyz";
const char *pw_symbols = "!\"#$%&'()*+,-./:;<=>?@@[\\]^_`{|}~";
const char *pw_ambiguous = "B8G6I1l0OQDS5Z2";
```
Remove the ambiguous characters from the set.

Or you can rely on the fact that the password will be copy & pasted anyways.

File: api/src/main/java/com/gentics/mesh/util/PasswordUtil.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+	/**
+	 * Generate human readable password with given length.
+	 * 
+	 * @param len
+	 * @return
+	 */
+	public static String humanPassword(int len) {
+		StringBuilder sb = new StringBuilder(len);
+		for (int i = 0; i < len; i++) {
+			int index = new Random().nextInt(ALPHA_NUMERIC_READABLE_CHAR.length());
```
Comment (philippguertler) [inline api/src/main/java/com/gentics/mesh/util/PasswordUtil.java]: Create the random object only once

File: api/src/main/java/com/gentics/mesh/util/PasswordUtil.java [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+/**
+ * Utility class which provides methods to handle passwords.
+ */
+public final class PasswordUtil {
+
+	private static final String ALPHA_NUMERIC_READABLE_CHAR = "ABCDEFGHJKLMNPQRTUVWXYZ"
+		+ "2346789"
+		+ "abcdefghjkmnpqrtuvxyz";
+
+	private static final int DEFAULT_PW_LEN = 6;
```
Comment (philippguertler) [inline api/src/main/java/com/gentics/mesh/util/PasswordUtil.java]: Why 6? For a limited set of characters, more then 10 should be the default length.

File: api/src/test/java/com/gentics/mesh/util/PasswordUtilTest.java [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+
+import org.junit.Test;
+
+public class PasswordUtilTest {
+
+	@Test
+	public void testHumanPassword() {
+		String pw = PasswordUtil.humanPassword(10);
+		assertEquals(pw.length(), 10);
+		System.out.println(pw);
```
Comment (philippguertler) [inline api/src/test/java/com/gentics/mesh/util/PasswordUtilTest.java]:
```suggestion
```

File: api/src/main/java/com/gentics/mesh/util/PasswordUtil.java (hunk 2 above)
Comment (philippguertler) [reply api/src/main/java/com/gentics/mesh/util/PasswordUtil.java]: Also, consider using SecureRandom

File: api/src/test/java/com/gentics/mesh/util/PasswordUtilTest.java [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+	}
+
+	@Test
+	public void testExcludedChars() {
+		List<Character> disallowedChars = Arrays.asList('0', 'O', 'o', '1', 'i', 'I', '5', 'S', 's');
+		for (int i = 0; i < 10000; i++) {
+			String pw = PasswordUtil.humanPassword(20);
+			for (int e = 0; e < pw.length(); e++) {
+				char pwChar = pw.charAt(e);
+				assertFalse("Found disallowed character in pw {" + pw + "} - {" +  pwChar+ "}", disallowedChars.contains(pwChar));
```
Comment (philippguertler) [inline api/src/test/java/com/gentics/mesh/util/PasswordUtilTest.java]:
```suggestion
				assertFalse("Found disallowed character in pw {" + pw + "} - {" + pwChar + "}", disallowedChars.contains(pwChar));
```

File: api/src/main/java/com/gentics/mesh/util/PasswordUtil.java [hunk 6]
```diff
@@ -1,17 +1,19 @@
...
 /**
  * Utility class which provides methods to handle passwords.
  */
 public final class PasswordUtil {
 
-	private static final String ALPHA_NUMERIC_READABLE_CHAR = "ABCDEFGHJKLMNPQRTUVWXYZ"
-		+ "2346789"
-		+ "abcdefghjkmnpqrtuvxyz";
+	private static final String ALPHA_NUMERIC_READABLE_CHAR = "ACEFGHJKLMNPRTUVWXY"
+		+ "234679"
```
Comment (philippguertler) [inline api/src/main/java/com/gentics/mesh/util/PasswordUtil.java]: G6

**Pull request merged**